ethash/src/compute.rs

```diff
@@ -391,9 +434,11 @@ fn light_new<T: AsRef<Path>>(cache_dir: T, block_number: u64) -> Light {
 	assert!(cache_size % NODE_BYTES == 0, "Unaligned cache size");
 	let num_nodes = cache_size / NODE_BYTES;
 
-	let mut nodes = Vec::with_capacity(num_nodes);
-	nodes.resize(num_nodes, Node::default());
+	let mut nodes: Vec<Node> = Vec::with_capacity(num_nodes);
 	unsafe {
+		// Use uninit instead of unnecessarily writing `size_of::<Node>() * num_nodes` 0s
+		nodes.set_len(num_nodes);
```

Comment:
the loop that sets the `bytes` of each node to the hash of the previous' will be broken now if `nodes[0].bytes` is anything but default.